service/src/main/java/bio/terra/workspace/service/petserviceaccount/PetSaService.java

```diff
@@ -79,27 +80,41 @@ public Policy enablePetServiceAccountImpersonation(
    *
    * @param workspaceUuid ID of the workspace to enable pet SA in
    * @param userToEnableEmail The user whose proxy group will be granted permission.
-   * @param token Token for calling SAM.
+   * @param userReq Auth token for calling SAM.
    * @param eTag GCP eTag which must match the pet SA's current policy. If null, this is ignored.
    * @return The new IAM policy on the user's pet service account, or empty if the eTag value
    *     provided is non-null and does not match current IAM policy on the pet SA.
    */
   public Optional<Policy> enablePetServiceAccountImpersonationWithEtag(
-      UUID workspaceUuid, String userToEnableEmail, String token, @Nullable String eTag) {
-    String petSaEmail =
-        SamRethrow.onInterrupted(
-            () ->
-                samService.getOrCreatePetSaEmail(
-                    gcpCloudContextService.getRequiredGcpProject(workspaceUuid), token),
-            "enablePet");
+      UUID workspaceUuid,
+      String userToEnableEmail,
+      AuthenticatedUserRequest userReq,
```

Comment:
nit: I can't tell if this change to passing `userReq` is aesthetic or functional. It doesn't matter; just curious. The comment on line 83 is maybe out of date as a result.

Reply:
mostly aesthetic, we generally pass `AuthenticatedUserReqest` objects around instead of tokens directly.

the only functional concern is avoiding potential confusion with `userReq.getEmail()`, which will be the caller's email (but might be a pet email instead of an end user email, while `userToEnableEmail` and `petSaName` are guaranteed to be a user and pet email, respectively). Added more to the javadoc to help prevent a future mistake like that

Reply:
SGTM, thanks!